test/framework/src/main/java/org/elasticsearch/test/AbstractStreamableXContentTestCase.java

```diff
@@ -51,6 +51,15 @@ protected boolean supportsUnknownFields() {
         return true;
     }
 
+    /**
+     * Indicates whether the parser supports comparing XContent after serialization and deserialization.
+     * In case it does, such behaviour will be tested by comparing the XContent of the original instance
+     * and the parsed/deserialized instance.
+     */
+    protected boolean assertToXContentEquivalence() {
```

Comment:
:+1:

Reply:
@nik9000 So what do you reckon? I shouldn't extend `AbstractStreamableXContentTestCase`?

Reply:
I think you can extend that to get good randomized coverage for the case when you don't have any exceptions and then hand write a couple of test cases for the exception round tripping. I *think* @javanna talked about doing that some other place....

Reply:
Have a look at `ListTasksResponseTests`, where we separate testing between two scenarios: without (no big problems, assertToXContentEquivalence true) and with (assertToXContentEquivalence false). I think that this change may not be needed if you do it that way.